spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/ModuleKey.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core;
+
+import java.io.Serializable;
+
+import org.springframework.util.Assert;
+
+/**
+ * Unique identifier for a {@link ModuleDeploymentRequest}.
+ * Methods {@link #toString()} and {@link #parse(String)} can be used to convert
+ * a key to a string and a string to a key, respectively. The id string may be
+ * used to uniquely identify a module in a database or execution environment.
+ *
+ * @author Patrick Peralta
+ */
+public class ModuleKey implements Serializable {
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * The name of the associated group or deployment unit this
+	 * module belongs to, such as a stream or job.
```

Comment:
remove "job" here also... for all of these cases, it might be ok to just use "group" and/or "stream" (deployment unit may not be a term we need anymore if this no longer applies to jobs)